### PR TITLE
Make description fields non-optional in the API

### LINKED ISSUE #828: Description for ServiceClass and ServicePlan should be required
...and not optional, as they seem to be by virtue of being pointers.  Instead these fields should just be strings.

### PR DESCRIPTION
**Pull request by pmorie**
Closes #828

### CODE REVIEW

File: pkg/apis/servicecatalog/v1alpha1/types.go [hunk 1]
```diff
@@ -149,7 +149,7 @@ type ServicePlan struct {
 	// CLI-friendly name of this plan
 	Name string `json:"name"`
 
-	Description *string `json:"description,omitempty"`
+	Description string `json:"description,omitempty"`
```
Comment (MHBauer) [inline pkg/apis/servicecatalog/v1alpha1/types.go]: what about omit empty here?

Comment (pmorie) [inline pkg/apis/servicecatalog/v1alpha1/types.go]: good catch

**Comment by pmorie** [comment]
cc @vaikas-google

**Comment by MHBauer** [comment]
Please put the generated code in a separate commit in the future.

File: pkg/apis/servicecatalog/validation/serviceclass_test.go [hunk 2]
```diff
@@ -24,171 +24,124 @@ import (
 	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
 )
 
+func validServiceClass() *servicecatalog.ServiceClass {
```
Comment (MHBauer) [inline pkg/apis/servicecatalog/validation/serviceclass_test.go]: Is implementing a gofuzz fuzzer a good idea for this kind of thing?

File: pkg/controller/controller.go [hunk 3]
```diff
@@ -1718,13 +1717,11 @@ func convertCatalog(in *brokerapi.Catalog) ([]*v1alpha1.ServiceClass, error) {
...
-		descCopy := string(plan.Description)
 		ret[i] = v1alpha1.ServicePlan{
-			Name:    plan.Name,
-			OSBGUID: plan.ID,
-			// OSBMetadata: plan.Metadata,
+			Name:        plan.Name,
+			OSBGUID:     plan.ID,
 			OSBFree:     plan.Free,
-			Description: &descCopy,
+			Description: plan.Description,
```
Comment (MHBauer) [inline pkg/controller/controller.go]: now we definitely don't have #742 to worry about.

File: pkg/apis/servicecatalog/validation/serviceclass_test.go (hunk 2 above)
Comment (pmorie) [inline pkg/apis/servicecatalog/validation/serviceclass_test.go]: Not sure what you mean - i think the default fuzzer is fine for this field. Were you asking if we need a special fuzzer for these types?

**Pull request merged**